AWSPinpoint/AWSPinpointEndpointProfile.m

```diff
@@ -538,12 +555,78 @@ + (BOOL)supportsSecureCoding {
 - (id)initWithCoder:(NSCoder *)decoder {
     if (self = [super init]) {
         _userId = [decoder decodeObjectOfClass:[NSString class] forKey:@"userId"];
+        _userAttributes = [decoder decodeObjectOfClass:[NSDictionary class] forKey:@"userAttributes"];
     }
     return self;
 }
 
 - (void)encodeWithCoder:(NSCoder *)encoder {
     [encoder encodeObject:_userId forKey:@"userId"];
+    [encoder encodeObject:_userAttributes forKey:@"userAttributes"];
+}
+
+- (void)addUserAttribute:(NSArray *)theValue
+                  forKey:(NSString *)theKey {
+    if (!theKey) {
+        AWSDDLogWarn(@"The key of user attribute you tried to add is empty.");
+        return;
+    }
+    
+    if (!self.userAttributes) {
+        self.userAttributes = [NSMutableDictionary dictionary];
+    }
+    
+    [self.userAttributes setValue:[AWSPinpointEndpointProfileUser processUserAttributeValues:theValue]
+                           forKey:[AWSPinpointEndpointProfileUser trimKey:theKey
+                                                                  forType:@"userAttribute"]];
+}
+
+- (NSDictionary *)allUserAttributes {
+    return [NSDictionary dictionaryWithDictionary:self.userAttributes];
+}
+
+- (BOOL)hasUserAttributeForKey:(NSString *)theKey {
+    return [self.userAttributes objectForKey:theKey] != nil;
+}
```

Comment:
Technically, i think this would be more correct:
```
    NSArray *items = [self.userAttributes objectForKey:theKey];
    return items.count;
```
because customers could call:
```
        user.addAttribute([], forKey: "sampleKey")
```
but... hm lets keep it as is to be consistent with the other `addAttribute`